docs/Module - CI.md

```diff
@@ -1,26 +1,112 @@
-    permissions:
-      contents: write
-      pull-requests: read
-```
+# Module CI - Workflow Overview
+
+## Module CI 
+ File : [`.github/workflows/module-ci.yaml`](https://github.com/cloudeteer/terraform-governance/blob/main/.github/workflows/module-ci.yaml)
+
+### Purpose 
+The `module-ci` workflow orchestrates multiple modular workflows for validation, linting, documentation, code analysis, and testing. Additionally, it automates the creation of GitHub issues for failed scheduled workflows.
+
+### Triggers 
+This workflow is triggered using `workflow_call` to allow reuse by other workflows.  
+
+It requires the following secrets:
+- `ARM_CLIENT_ID`: Azure client ID
+- `ARM_SUBSCRIPTION_ID`: Azure subscription ID
+- `ARM_TENANT_ID`: Azure tenant ID
+
+### Jobs 
+1.  Validate : Ensures Terraform configurations are valid by running the [`module-ci-validate.yaml`](https://github.com/cloudeteer/terraform-governance/blob/main/.github/workflows/module-ci-validate.yaml) workflow
+2.  Lint : Enforces best practices using the [`module-ci-lint.yaml`](https://github.com/cloudeteer/terraform-governance/blob/main/.github/workflows/module-ci-lint.yaml) workflow
+3.  Documentation : Generates and validates Terraform module documentation using the [`module-ci-documentation.yaml`](https://github.com/cloudeteer/terraform-governance/blob/main/.github/workflows/module-ci-documentation.yaml) workflow
+4.  Code Analysis : Scans for misconfigurations and secrets using the [`module-ci-code-analysis.yaml`](https://github.com/cloudeteer/terraform-governance/blob/main/.github/workflows/module-ci-code-analysis.yaml) workflow
+5.  Test : Runs Terraform tests in various environments via the [`module-ci-test.yaml`](https://github.com/cloudeteer/terraform-governance/blob/main/.github/workflows/module-ci-test.yaml) workflow
+6.  Issue Creation : Creates GitHub issues for failed workflows by calling the [`module-ci-issue.yaml`](https://github.com/cloudeteer/terraform-governance/blob/main/.github/workflows/module-ci-issue.yaml) workflow
+
+---
+
+## Module CI Validate 
+ File : [`.github/workflows/module-ci-validate.yaml`](https://github.com/cloudeteer/terraform-governance/blob/main/.github/workflows/module-ci-validate.yaml)
+
+### Purpose 
+This workflow validates Terraform configurations to ensure they are syntactically correct, formatted properly, and meet validation standards.
+
+### Steps 
+1.  Repository Checkout : Clones the repository for validation
+2.  Fetch Terraform Version : Determines the required Terraform version using a custom action
+3.  Terraform Setup : Installs and configures Terraform based on the required version
+4.  Terraform Initialization : Runs `terraform init` with `-backend=false` to skip backend configuration
+5.  Format Check : Ensures consistent code style using `terraform fmt`
+6.  Validation : Checks the validity of Terraform configurations using `terraform validate`
```

Comment:
I believe this provides too much detail (both here and in the subsequent `### Steps` sections). There’s no need to document every step a job executes. Doing so would require updating the documentation every time the execution steps change, which creates unnecessary overhead. If someone wants to know the details of each step, they can simply open the workflow YAML file and review it there.